src/components/screens/timebank-screen.tsx

```diff
@@ -0,0 +1,118 @@
+import { useEffect, useState } from "react";
+import { Person, Timespan } from "../../generated/client";
+import { CircularProgress, Card } from "@mui/material";
+import { userProfileAtom } from "../../atoms/auth";
+import { useAtom, useAtomValue, useSetAtom } from "jotai";
+import { errorAtom } from "../../atoms/error";
+import {
+  dailyEntriesAtom,
+  personDailyEntryAtom,
+  personTotalTimeAtom,
+  personsAtom
+} from "../../atoms/person";
+import { useApi } from "../../hooks/use-api";
+import TimebankContent from "../timebank/timebank-content";
+import { DateTime } from "luxon";
+import strings from "../../localization/strings";
+import config from "../../app/config";
+
+/**
+ * Time bank screen component.
+ */
+const TimebankScreen = () => {
+  const userProfile = useAtomValue(userProfileAtom);
+  const [timespanSelector, setTimespanSelector] = useState<Timespan>(Timespan.ALL_TIME);
+  const setError = useSetAtom(errorAtom);
+  const { personsApi, dailyEntriesApi } = useApi();
+  const persons = useAtomValue(personsAtom);
+  const [loading, setLoading] = useState(false);
+  const [personTotalTime, setPersonTotalTime] = useAtom(personTotalTimeAtom);
+  const [personDailyEntry, setPersonDailyEntry] = useAtom(personDailyEntryAtom);
+  const [dailyEntries, setDailyEntries] = useAtom(dailyEntriesAtom);
+
+  useEffect(() => {
+    getPersonTotalTime();
+    getPersonDailyEntries();
+  }, [persons]);
+
+  /**
+   * Gets person's total time data.
+   *
+   * @param timespan enum
+   */
+  const getPersonTotalTime = async (timespan?: Timespan) => {
+    setLoading(true);
+    if (!personTotalTime || timespan) {
+      setTimespanSelector(timespan || Timespan.ALL_TIME);
```

Comment:
We had a discussion about this. The timespan state should be handled via a handler tied to the select onchange. The span related code can then be removed from this block. Timespan can be added to the dependency list of the UseEffect that calls this function and then there will be some refactoring required related to the conditionals here. As I understand it this function has the purpose of both updating the persons total time when the timespan is changed and also initializing the personTotalTime atom in the case a user navigates directly to the timebank screen without using the dashboard screen. Perhaps these cases should be separated into separate functions.

Reply:
Turns out there's no need to check for conditions to run the persontotaltime fetch after the null return, the code runs nicely even if you were to navigate straight to the timebank route.

I've removed the call of this function (getPersonTotalTime) from the Select onChange handler (which is in the child component timebank-content) as well removed the timespan parameter and replaced it with timespanSelector useState.

The Select menu onChange event now changes the timespanSelector state, which is now a dependency in the aforementioned useEffect, which triggers this function.